test/support/utils_test_base.h

```diff
@@ -0,0 +1,695 @@
+        template<typename _Iterator, typename TDiff>
+        void update_data(_Iterator __it, TDiff __objects_count)
+        {
+            src_data_usm.update_data(__it, offset, __objects_count);
+        }
+    };
+    ::std::vector<Data> data;   // Vector of source test data:
+                                //  - 1 item for test1buffer;
+                                //  - 2 items for test2buffers;
+                                //  - 3 items for test3buffers
+
+    test_base_data_usm(sycl::queue __q, InitParams init);
+
+    TestValueType* get_start_from(UDTKind kind);
+
+// test_base_data
+
+    // Check that host buffering is required
+    virtual bool host_buffering_required() const override;
+
+    // Get test data
+    virtual TestValueType* get_data(UDTKind kind) override;
+
+    // Retrieve data
+    virtual void retrieve_data(UDTKind kind, TestValueType* __it_from, TestValueType* __it_to) override;
+
+    // Update data
+    virtual void update_data(UDTKind kind, TestValueType* __it_from, TestValueType* __it_to) override;
+};
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+#if TEST_DPCPP_BACKEND_PRESENT
+////////////////////////////////////////////////////////////////////////////////
+/// struct test_base_data_buffer - test source data for SYCL buffer
+template <typename TestValueType>
+struct test_base_data_buffer : test_base_data<TestValueType>
+{
+    struct Data
+    {
+        using TSourceData = sycl::buffer<TestValueType, 1>;
+
+        TSourceData   src_data_buf;     // SYCL buffer
+        ::std::size_t offset = 0;       // Offset in SYCL buffer
+
+        template<typename _Size>
+        Data(_Size __sz, ::std::size_t __offset)
+            : src_data_buf(sycl::range<1>(__sz + __offset))
+            , offset(__offset)
+        {
+        }
+    };
+    ::std::vector<Data> data;   // Vector of source test data:
+                                //  - 1 item for test1buffer;
+                                //  - 2 items for test2buffers;
+                                //  - 3 items for test3buffers
+
+    test_base_data_buffer(InitParams init);
+
+    sycl::buffer<TestValueType, 1>& get_buffer(UDTKind kind);
+
+    auto get_start_from(UDTKind kind)
+        -> decltype(oneapi::dpl::begin(data.at(enum_val_to_index(kind)).src_data_buf));
+
+// test_base_data
+
+    // Check that host buffering is required
+    virtual bool host_buffering_required() const override;
+
+    // Get test data
+    virtual TestValueType* get_data(UDTKind kind) override;
+
+    // Retrieve data
+    virtual void retrieve_data(UDTKind kind, TestValueType* __it_from, TestValueType* __it_to) override;
+
+    // Update data
+    virtual void update_data(UDTKind kind, TestValueType* __it_from, TestValueType* __it_to) override;
+};
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+////////////////////////////////////////////////////////////////////////////////
+/// struct test_base_data_sequence -  test source data for sequence (based on std::vector)
```

Comment:
//TODO: consider remove 'struct test_base_data_sequence' (and usage of) because it does nothing.

Reply:
Done